### ISSUE TITLE
Handling feed updates for manipulated entries

### ISSUE THREAD

**Issue by kevinpapst** [FreshRSS/FreshRSS#1706]
Hi guys,
sorry for the disturbing title, I couldn't come up with a better one ;-)

Here is my problem:
After submitting the Dilbert extension, I recognized that there seems to be a strange behaviour when updating the feed.

When add new entries, the extensions rewrites the author and title:
https://github.com/kevinpapst/freshrss-dilbert/blob/master/xExtension-Dilbert/extension.php#L63

After the Cron updated the feed (this one is updated on a daily base) all entries from the feed were marked as unread and title and content changed back to the original feed version:

<img width="736" alt="bildschirmfoto 2017-12-03 um 14 19 52" src="https://user-images.githubusercontent.com/533162/33525690-17c4de26-d835-11e7-8b1b-6ae2f200091a.png">

The feed contains 7 entries, so the newest one was added correctly and the oldest one was kept as is, likely because it wasn't included in the feed anymore. The 6 other entries were switched back to their original state (title, author and read status). I could imagine that the update process somehow flips the entries content without calling the "entry_before_insert" hook => but why?

I cannot reproduce this behaviour with the "Actualize" button, maybe because there is no new feed entry.

Any idea what is going on here?
Which fields are used for the duplicate entry detection?
Can you point me in a code direction, so I can have a look whats going on?

**Comment by Alkarex** [comment]
Hum, indeed, we have to keep the hash of the unmodified entry.

https://github.com/FreshRSS/FreshRSS/blob/4e174ed9dd84ce4f8de410baf6a6e0fde9160055/app/Models/Entry.php#L92-L98

https://github.com/FreshRSS/FreshRSS/blob/28db5dd2858002b376c920ec1d1aaad4ef254d54/app/Controllers/feedController.php#L322-L350

Let me have a look.

**Comment by kevinpapst** [comment]
Understand, it sees the entry as updated and changes the content back. 
Two quick ideas without checking the code:
- introduce a hook to calculate/overwrite the entries hash in the extension, then I could easily handle it myself
- calculate the hash from the original feed entry and not from the one that comes back from the extension ("entry_before_insert" hook)

**Comment by Alkarex** [comment]
Please see / test https://github.com/FreshRSS/FreshRSS/pull/1707
In your extension callback function:

```php
$originalHash = $entry->hash();
// your code
$entry->_hash($originalHash);
```

**Comment by kevinpapst** [comment]
Any idea on how to test that now? Using the "actualize" button didn't do anything before.
Can I reset the last updated status in the database on that feed?

Edit:
By manipulating the lastUpdate column and deleting single entries from the database, then updating I could reproduce the problem.
After making the adaptions and re-adding the feed the problem was gone!

Thanks for the quick help @Alkarex ! Shall I close the ticket or will it be closed when 1.9 is released?
